nise/generators/aws/ec2_generator.py

```diff
@@ -94,6 +111,8 @@ def __init__(self, start_date, end_date, currency, payer_account, usage_accounts
                 self._product_sku = self.attributes.get("product_sku")
             if self.attributes.get("tags"):
                 self._tags = self.attributes.get("tags")
+            if self.attributes.get("operating_system"):
+                self._tags = self.attributes.get("operating_system")
```

Comment:
This is supposed to assign to `self._operation_system`
```suggestion
            if self.attributes.get("operating_system"):
                self._operating_system = self.attributes.get("operating_system")
```

`self.attributes` is initialized in the parent class. This whole `if attributes` conditional block can be removed. As an example:
```python
        self._processor_arch = choice(self.ARCHS)
        self._resource_id = "i-{}".format(self.fake.ean8())
        self._product_sku = self.fake.pystr(min_chars=12, max_chars=12).upper()
        self._instance_type = choice(self.INSTANCE_TYPES)
        self._processor_arch = self.attributes.get("processor_arch")
        self._resource_id = "i-{}".format(self.attributes.get("resource_id"))
        self._product_sku = self.attributes.get("product_sku")
        self._tags = self.attributes.get("tags")

        if instance_type := self.attributes.get("instance_type"):
            self._instance_type = (
                instance_type.get("inst_type"),
                instance_type.get("physical_cores"),
                instance_type.get("vcpu"),
                instance_type.get("memory"),
                instance_type.get("storage"),
                instance_type.get("family"),
                instance_type.get("cost"),
                instance_type.get("rate"),
                instance_type.get("saving"),
                "${} per On Demand Linux {} Instance Hour",
            )
```

The instance attributes will be different by removing that conditional block, but it's better for objects to have consistent attributes with default values.

Reply:
It's less error prone to use named values in format string templates instead of positional parameters.
```python
description = "${cost} per On Demand Linux {inst_type} Instance Hour"

# Later on
    def _update_data(self, row, start, end, **kwargs):
        """Update data with generator specific data."""
        inst_type, physical_cores, vcpu, memory, storage, family, cost, rate, saving, description = self._instance_type
        inst_description = description.format(cost=cost, inst_type=inst_type)
```

Reply:
Thanks. That makes sense. I have modified the test to use named values.